### DISCUSSION TITLE
Apply Dirichlet bcs on the rotations (first derivatives) when using an Hermite FE

### DISCUSSION THREAD

**Discussion by StefanoPizzamiglio** [firedrakeproject/firedrake#2975] [Firedrake support] [answered]
Hi, 
I'm trying to solve the Euler Bernoulli beam bending problem using firedrake and Hermite elements. The variational formulation is:
a=div(grad(u))*EJ*div(grad(v))*dx
L=f*v*dx

However, I can't understand how to apply essential BCs on the rotations. If I use DirichletBC it only constraints the displacement u, not the associated rotation du/dx, which is the other DOF of the Hermite element. 

Can anyone help me?

**Comment by wence-** [comment, marked as answer]
Is this in 1D? I think it must be because we don't support essential boundary conditions on Hermite elements in 2D.

In 1D, we support setting the vertex value $u$, but not its derivative `$u_x$, on the boundary (this is implemented with a hack in `DirichletBC`)

https://github.com/firedrakeproject/firedrake/blob/c5cd37ad718da017f3035a5a7c6f5329d60e62aa/firedrake/bcs.py#L133-L138

So I think right now your best bet is to try and apply your BCs using Nitsche's method instead.
